src/main/webapp/WEB-INF/web.xml

```diff
@@ -1,17 +1,27 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_4_0.xsd"
-         version="4.0">
 
-    <servlet>
-        <servlet-name>GenericServlet</servlet-name>
-        <servlet-class>Controllers.GenericServlet</servlet-class>
-    </servlet>
-    <servlet-mapping>
-        <servlet-name>GenericServlet</servlet-name>
-        <url-pattern>/</url-pattern>
-    </servlet-mapping>
+<web-app xmlns="http://java.sun.com/xml/ns/javaee"
+	 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	 xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd"
+	 version="3.0">
+    
+    <filter>
+        <filter-name>LoginFilter</filter-name>
+        <filter-class>Filters.LoginFilter</filter-class>
+    </filter>
+    <filter-mapping>
+        <filter-name>LoginFilter</filter-name>
+        <url-pattern>/*</url-pattern>
+    </filter-mapping>
+
+    <filter>
+        <filter-name>DashboardFilter</filter-name>
+        <filter-class>Filters.DashboardFilter</filter-class>
+    </filter>
+    <filter-mapping>
+        <filter-name>DashboardFilter</filter-name>
+        <url-pattern>/dashboard/*</url-pattern>
```

Comment:
Why do we need a `*` here but not on the `RegistrationServlet` URL pattern?

Reply:
Hmm, not really sure to be honest - it should really be set up to redirect all /dashboard/* stuff to just /dashboard, though it currently just gives you the right page but for weird urls like /dashboard/weirdurls.
![image](https://user-images.githubusercontent.com/26003386/102026569-83c44980-3d96-11eb-9e1a-dd09179e4679.png)